src/libraries/Native/Unix/System.Native/pal_console.c

```diff
@@ -180,7 +180,7 @@ static bool ConfigureTerminal(bool signalForBreak, bool forChild, uint8_t minCha
 
     if (!forChild)
     {
-        termios.c_iflag &= (uint32_t)(~(IXON | IXOFF));
+        termios.c_iflag &= (uint32_t)(~(IXON | IXOFF | ICRNL | INLCR | IGNCR));
```

Comment:
This will affect any read.
Which seems ok, because everything ends up going through `ReadKeyCore(bool convertCrToLf` which handles proper conversion.
The only exception is `Console.OpenStandardInput()` which will no longer have `\r` converted to `\n`. That is fine for me.

Reply:
> The only exception is Console.OpenStandardInput() which will no longer have \r converted to \n.

I added the breaking change label for this.

Reply:
Hmm, it looks like `UnixConsoleStream.Read()` will read directly from stdin without touching any of the `console_pal` apis. Wouldn't this mean that the input will vary depending on whether other console methods have hit the `ConfigureTerminal` method?

Reply:
> Wouldn't this mean that the input will vary depending on whether other console methods have hit the ConfigureTerminal method?

Yes. You could `InitializeConsoleBeforeRead` as part of `OpenStandardInput` to consistently configure the terminal.
```cs
public static Stream OpenStandardInput()
{
    Interop.Sys.InitializeConsoleBeforeRead();
    try
    {
        return new UnixConsoleStream(SafeFileHandleHelper.Open(() => Interop.Sys.Dup(Interop.Sys.FileDescriptors.STDIN_FILENO)), FileAccess.Read);
    }
    finally
    {
        Interop.Sys.UninitializeConsoleAfterRead();
    }
}
```

Reply:
Is there any reason why we should guard the constructor with a finally clause? I was thinking I could just include
```
Interop.Sys.InitializeConsoleBeforeRead();
Interop.Sys.UninitializeConsoleAfterRead();
```
inside the `UnixConsoleStream` constructor.